src/effects/VST3/VST3Effect.cpp

```diff
@@ -242,170 +237,6 @@ bool VST3Effect::LoadFactoryPreset(int id, EffectSettings& settings) const
    return true;
 }
 
-namespace
-{
-   unsigned CountChannels(Steinberg::Vst::IComponent* component,
-      const Steinberg::Vst::MediaTypes mediaType, 
-      const Steinberg::Vst::BusDirection busDirection,
-      const Steinberg::Vst::BusType busType)
-   {
-      using namespace Steinberg;
-
-      unsigned channelsCount{0};
-
-      const auto busCount = component->getBusCount(mediaType, busDirection);
-      for(auto i = 0; i < busCount; ++i)
-      {
-         Vst::BusInfo busInfo;
-         if(component->getBusInfo(mediaType, busDirection, i, busInfo) == kResultOk)
-         {
-            if(busInfo.busType == busType)
-               channelsCount += busInfo.channelCount;
-         }
-      }
-      return channelsCount;
-   }
-}
-
-unsigned VST3Effect::GetAudioInCount() const
-{
-   return CountChannels(
-      mWrapper->mEffectComponent,
-      Steinberg::Vst::kAudio,
-      Steinberg::Vst::kInput,
-      Steinberg::Vst::kMain);
-}
-
-unsigned VST3Effect::GetAudioOutCount() const
-{
-   return CountChannels(
-      mWrapper->mEffectComponent,
-      Steinberg::Vst::kAudio,
-      Steinberg::Vst::kOutput,
-      Steinberg::Vst::kMain);
-}
-size_t VST3Effect::SetBlockSize(size_t maxBlockSize)
-{
-   mProcessingBlockSize = 
-      static_cast<Steinberg::int32>(std::min(maxBlockSize, mUserBlockSize));
-   return mProcessingBlockSize;
-}
-
-size_t VST3Effect::GetBlockSize() const
-{
-   return mProcessingBlockSize;
-}
-
-sampleCount VST3Effect::GetLatency() const
-{
-   if(mUseLatency)
-   {
-      if(!mRealtimeGroupProcessors.empty())
-         return mRealtimeGroupProcessors[0]->GetLatencySamples();
```

Comment:
This deleted line has no corresponding one in the new function for the Instance.  But I see from other functions that you assume latency is available at initialization time and does not change, so you cache the value.

Ok.

But I wonder if varying latency, with varying settings, really is a possible thing we will need to handle later.  You know that fixing the handling of latency in realtime play is a problem I need to solve, maybe in a patch release.

Reply:
In case of VST3 plugin should explicitly ask host to restart (via setActive) effect component  by sending IComponentHanler::restartComponent(kLatencyChanged).